packages/big-design/src/components/Table/Table.tsx

```diff
@@ -36,31 +44,34 @@ const InternalTable = <T extends TableItem>(props: TableProps<T>): React.ReactEl
             return;
           }
 
-          const { selectedItems, onSelectionChange } = selectable;
-          const nextIsSelected = !isItemSelected(item);
+          const { onSelectionChange } = selectable;
+          const nextIsSelected = !selectedItems.has(item);
 
           if (nextIsSelected) {
             onSelectionChange([...selectedItems, item]);
           } else {
-            onSelectionChange(selectedItems.filter(selectedItem => selectedItem !== item));
+            onSelectionChange([...selectedItems].filter(selectedItem => selectedItem !== item));
```

Comment:
`.filter` should return a new Array, any reason you destructed it?

Reply:
`selectedItems` in this case is a `Set` so I'm converting it to an array and filtering it.